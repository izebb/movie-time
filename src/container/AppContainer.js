import React from 'react';

import { useMovies } from './movieHook';

import { MovieList } from '../components/MovieList';
import { MovieHeader } from '../components/MovieHeader';
import { Preloader } from '../components/Preloader';
import { Error } from '../components/Error';

const initialState = {
  movieList: [],
  isFetching: false,
  displaySection: {},
  error: null
};

export function AppContainer() {
  const [moviesState, handlers] = useMovies(initialState);
  const {
    movieList,
    displaySection,
    displaSectionIndex,
    isFetching,
    error
  } = moviesState;
  const { displaySummary, clearMovieSelection, tryAgain } = handlers;

  return (
    <>
      <MovieHeader />
      {error ? (
        <Error message={error} retry={tryAgain} />
      ) : isFetching ? (
        <Preloader />
      ) : (
        <MovieList
          displaySection={displaySection}
          displaSectionIndex={displaSectionIndex}
          list={movieList}
          onClickMovie={displaySummary}
          clearMovieSelection={clearMovieSelection}
        />
      )}
    </>
  );
}
