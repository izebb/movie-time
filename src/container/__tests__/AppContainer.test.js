import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AppContainer } from '../AppContainer';

jest.useFakeTimers();
describe('AppContainer', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AppContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
