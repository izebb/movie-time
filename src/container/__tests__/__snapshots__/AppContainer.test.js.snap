// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should render without crashing 1`] = `
<Fragment>
  <MovieHeader />
  <MovieList
    clearMovieSelection={[Function]}
    displaSectionIndex={-1}
    displaySection={Object {}}
    list={Array []}
    onClickMovie={[Function]}
  />
</Fragment>
`;
